server/svix-server/src/core/types/mod.rs

```diff
@@ -437,6 +437,26 @@ impl Validate for EventChannelSet {
     }
 }
 
+string_wrapper!(EventTypeFeatureFlag);
+
+impl Validate for EventTypeFeatureFlag {
```

Comment:
I know we use this pattern a lot (relying on manually `Validate`ing), but I think a better pattern is just to forbid invalid types from being initialized at all.

`types::metadata` does this - where `serde::Deserialize` is manually implemented to just outright reject invalid values. This way, we don't have to worry about manually validating the type in the API definitions.

I think this would also let you avoid having to define a separate `EventTypeFeatureFlagSet`. We can just use `HashSet<EventTypeFeatureFlag>`, and not worry about invalid types still getting initialized.

Reply:
As discussed on slack split `string_wrapper!` into two macros to do this (because that did a `derive(Deserialize)`) and implemented `Deserialize` manually which calls `validate_limited_str`. Also, just made `FeatureFlagSet` a type  alias for `HashSet<FeatureFlag>`.